aptos-move/framework/aptos-framework/sources/configs/stake.move

```diff
@@ -860,24 +864,33 @@ module aptos_framework::stake {
 
     /// Update the number of missed votes. This is only called by block::prologue().
     /// This function cannot abort.
-    public(friend) fun update_performance_statistics(missed_votes: vector<u64>) acquires ValidatorPerformance {
+    public(friend) fun update_performance_statistics(proposer_index_optional: vector<u64>, failed_proposer_indices: vector<u64>) acquires ValidatorPerformance {
```

Comment:
I realized this is a vector just because `Move` doesn't have `Optional` support - is that the case? Adding a comment here explaining why this is optional and also why this is a vector would be helpful :)

Reply:
Move does have std::option. However the problem here is that you can't pass a struct (option::Option for example) in a direct invocation or a script function transaction (due to ownership concerns).